### PR TITLE
Update SDGym to work with SDV 1.0.0

### LINKED ISSUE #212: Cleanup SDGym to the new SDV 1.0 metadata and synthesizers
### Problem Description
Since we've upgraded to [SDV 1.0](https://docs.sdv.dev/sdv), the synthesizer class names and metadata formats have changed. The SDGym is currently written assuming that SDV < 1.0 (even though the official requirements are anything `>=0.18.0`). 

We should fix this asap.

### Expected behavior
Use the synthesizers, datasets and metadata as provided by the SDV 1.0 library. 

We should also update the minimum and maximum to: `sdv >1.0` and `sdv < 2.0`

### PR DESCRIPTION
**Pull request by fealho**
Resolve #212.

### CODE REVIEW

File: sdgym/benchmark.py [hunk 1]
```diff
@@ -133,12 +113,10 @@ def _compute_scores(metrics, real_data, synthetic_data, metadata, output, comput
             normalized_score = None
             start = datetime.utcnow()
             try:
-                LOGGER.info('Computing %s on dataset %s', metric_name, metadata._metadata['name'])
```
Comment (amontanez24) [inline sdgym/benchmark.py:136]: I think we can keep the logs

File: sdgym/benchmark.py [hunk 2]
```diff
@@ -151,75 +129,52 @@ def _compute_scores(metrics, real_data, synthetic_data, metadata, output, comput
 
     if compute_quality_score:
         start = datetime.utcnow()
-        if metadata.modality == 'single-table':
```
Comment (amontanez24) [inline sdgym/benchmark.py:154]: are we getting rid of the modality checks because this PR will drop support for other modalities?

Comment (fealho) [reply sdgym/benchmark.py:154]: In this PR I tried to make the minimal changes to make the code work. I didn't want to completely delete the support for other modalities here. I think if we want to do that it should be done in a different PR.

Another option is to not merge this PR to master, and instead to some other branch `ready-for-release`, which in turn will be merged to master. We can then make all the other incremental changes there, if you think merging code with pieces that don't work is a problem.

File: sdgym/benchmark.py (hunk 1 above)
Comment (fealho) [reply sdgym/benchmark.py:136]: I deleted them because we don't have the parameters anymore in most of them. I can try to fix that though.

Comment (amontanez24) [reply sdgym/benchmark.py:136]: yeah let's try to get them or something close. I think we could use more logging in our libraries lol

File: sdgym/synthesizers/base.py [hunk 3]
```diff
@@ -146,6 +146,8 @@ class MultiSingleTableBaselineSynthesizer(BaselineSynthesizer, abc.ABC):
 
     These classes model and sample each table independently and then just
     randomly choose ids from the parent tables to form the relationships.
+
+    NOTE: doesn't currently work.
```
Comment (pvk-developer) [inline sdgym/synthesizers/base.py:150]: Out of curiosity, did it work before ? If so do we know why is not currently working ?

**Review by pvk-developer — APPROVED** [review]
LGTM! Solid work 🪨

**Review by amontanez24 — APPROVED** [review]
LGTM! Thanks for the changes :shipit:

**Pull request merged**